src/mgr/ClusterState.cc

```diff
@@ -124,8 +113,11 @@ void ClusterState::ingest_pgstats(ref_t<MPGStats> stats)
 	       << q->second.reported_seq << dendl;
       continue;
     }
-
-    pending_inc.pg_stat_updates[pgid] = pg_stats;
+    if (pg_state_string(pg_stats.state) == "deleted") {
```

Comment:
why compare against the string, instead of the actual state?

Reply:
If one pg will be delete, we no set pg state on `PeeringState::Deleting::Deleting(my_context ctx)`. The reason for comparing here is entirely to remove the pg in the previous deleting state. Perhaps we can provided a new method `pg_state_test` in `osd_types.h`

Reply:
Let me rephrase the question:
why not
'if (pg_stats.state == <whatever the constant is for "deleted">) ...'

i.e. - why bother changing a constant to string?